src/Platforms/SQLServer2012Platform.php

```diff
@@ -1010,7 +1010,8 @@ public function getListTableForeignKeysSQL($table, $database = null)
                 INNER JOIN sys.objects AS o ON o.OBJECT_ID = fc.referenced_object_id
                 ON f.OBJECT_ID = fc.constraint_object_id
                 WHERE ' .
-                $this->getTableWhereClause($table, 'SCHEMA_NAME (f.schema_id)', 'OBJECT_NAME (f.parent_object_id)');
+                $this->getTableWhereClause($table, 'SCHEMA_NAME (f.schema_id)', 'OBJECT_NAME (f.parent_object_id)') .
+                ' ORDER BY fc.constraint_column_id';
```

Comment:
in SQLSrv some order bys add `ASC` and some don't, so I have not added it here to be in sync with your initial suggestion regarding MySQL. However, in the MySQL platform ASC is always there so I think we should add it there.